### PR TITLE
Extract LdapClient to plugin-toolkit

### PR DESCRIPTION
**Pull request by Praveen2112**
## Description
This extraction allows us to reuse the LdapClient across multiple modules. 

> Is this change a fix, improvement, new feature, refactoring, or other?

Refactoring around LdapClient

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Password authenticators

> How would you describe this change to a non-technical end user or system administrator?

Extract LdapClient to plugin toolkit 

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by aczajkowski — APPROVED** [review]
LGTM

**Review by kokosing — APPROVED** [review]

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/LdapQuery.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+        public LdapQueryBuilder setSearchFilter(String searchFilter)
+        {
+            this.searchFilter = searchFilter;
+            return this;
+        }
+
+        public LdapQueryBuilder setAttributes(String... attributes)
+        {
+            this.attributes = attributes;
```
Comment (lukasz-walkiewicz) [inline lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/LdapQuery.java:66]: `copyOf`?

File: plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticatorConfig.java [hunk 2]
```diff
@@ -34,7 +34,6 @@
     private String userBaseDistinguishedName;
     private String bindDistinguishedName;
     private String bindPassword;
-    private boolean ignoreReferrals;
```
Comment (lukasz-walkiewicz) [inline plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticatorConfig.java]: Remove in the previous commit

File: lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/LdapQuery.java (hunk 1 above)
Comment (s2lomon) [reply lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/LdapQuery.java:69]: + null handling

File: plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticator.java [hunk 3]
```diff
@@ -201,6 +226,19 @@ private String lookupUserDistinguishedName(String user)
         return getOnlyElement(userDistinguishedNames);
     }
 
+    private void validatePassword(String userName, String password)
```
Comment (s2lomon) [inline plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/LdapAuthenticator.java:229]: Just like the above - I like this kind of method and I think that this kind of abstraction should be provided for each of the `ldapClient` usages, regardless of the fact whether this method is used in one or more places. Preferably all this kind of methods should be extracted to a separate class. 

This should set proper boundaries in the code for using this kind of third party's service client and allow others to extend it in an organised manner.

**Review by s2lomon — APPROVED** [review]
lgtm %comments

**Review by s2lomon — APPROVED** [review]
Good job, I like how it's done.

**Pull request merged**